src/saml.ts

```diff
@@ -1074,37 +1074,43 @@ class SAML {
           }
         }
 
-        subjectConfirmation = subject[0].SubjectConfirmation
-          ? subject[0].SubjectConfirmation[0]
-          : null;
-        confirmData =
-          subjectConfirmation && subjectConfirmation.SubjectConfirmationData
-            ? subjectConfirmation.SubjectConfirmationData[0]
-            : null;
-        if (subject[0].SubjectConfirmation && subject[0].SubjectConfirmation.length > 1) {
-          msg = "Unable to process multiple SubjectConfirmations in SAML assertion";
-          throw new Error(msg);
-        }
+        if (subject[0].SubjectConfirmation && subject[0].SubjectConfirmation?.length > 0) {
```

Comment:
This doesn't seem to be doing what the spec calls for. The spec calls for "If multiple assertions are included, then each assertion's <Subject> element MUST refer to the same principal." However, I see no check for that. What is more, it seems that by use of `Array.find()`, you are allowing _some_ `SubjectConfirmation` to be valid instead of checking that they are _all_ valid.

Reply:
Sorry maybe I'm not so good at JS/TS for this code fix. But I'll try to do my best to fix as requested. Let me explain what the fix was meant for. I refer to this part "Verify that any assertions relied upon are valid in other respects. Note that while multiple bearer <SubjectConfirmation> elements may be present, the successful evaluation of a single such element in accordance with this profile is sufficient to confirm an assertion. However, each assertion, if more than one is present, MUST be evaluated independently."
That is related to [Issue 339](https://github.com/node-saml/passport-saml/issues/339) of passport-saml. I would also like to add a unit test for this change but I don't know how to get such a signing request. It's ok to use one from our internal test system?

Reply:
I don't think I have any particular object to you adding a test from your internal system as long as the license is compatible with this project and your employer and the code owner is OK with it. I'll have a look at it when it shows up and provide comments.